cpp/arcticdb/version/version_map.hpp

```diff
@@ -606,14 +608,9 @@ class VersionMapImpl {
                 return cached_timestamp <= requested_load_strategy.load_from_time_.value();
             }
             case LoadType::ALL:
```

Comment:
Sorry I didn't think about this before but currently if we do: `{LoadType::ALL, UNDELETED_ONLY}` repeatedly it would never get cached.

That is because if we have the following version chain:
```
v0 <- v1 <- tall1 <- v2
```

the load_all_undeleted would reach the tall0 and terminate there.

I think we already have the information we need. I.e. if `earliest_loaded_version <= tombstone_all.value()` then we know that we have all undeleted versions.

It would also be good to test this.

Reply:
Ah, good catch! Didn't think about that case